File: 프로그래머스/lv3/43238. 입국심사/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long l=1L,r=1000000000000000000L;
        while(l<r){
            long mid=(l+r)>>1;
            long now=0;
            for(int k:times){
                now+=mid/k;
            }
            if(now<n){
                l=mid+1;
            }
            else{
                r=mid;
            }
        }
        return l;
    }
}

File: 프로그래머스/lv3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = 0
    l,r=1,1000000000000000000
    while l<r:
        mid=(l+r)//2;
        now=0;
        for k in times:
            now+=mid//k
        if now<n:
            l=mid+1;
        else:
            r=mid;
    return l